test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs

```diff
@@ -0,0 +1,227 @@
+
+                    Assert.True(cursor.MoveNext());
+                    c = default;
+                    cgetter(ref c);
+
+                    Assert.Equal(2.0 * 3.0 + 2.0 * 3.0, c.Values[0]);
+                    Assert.Equal(2.0 * 3.0 + 2.0 * 3.0, c.Values[1]);
+                    Assert.Equal(2.0 * 3.0 + 2.0 * 3.0, c.Values[2]);
+                    Assert.Equal(2.0 * 3.0 + 2.0 * 3.0, c.Values[3]);
+
+                    Assert.False(cursor.MoveNext());
+
+                }
+            }
+        }
+
+        [Fact]
+        public void TensorFlowTransformMNISTConvTest()
+        {
+            var model_location = "mnist_model/frozen_saved_model.pb";
+            using (var env = new TlcEnvironment(seed: 1, conc: 1))
+            {
+                var dataPath = GetDataPath("Train-Tiny-28x28.txt");
+                var testDataPath = GetDataPath("MNIST.Test.tiny.txt");
+
+                // Pipeline
+                var loader = TextLoader.ReadFile(env,
+                new TextLoader.Arguments()
+                {
+                    Separator = "tab",
+                    HasHeader = true,
+                    Column = new[]
+                    {
+                        new TextLoader.Column()
+                        {
+                            Name = "Label",
+                            Source = new [] { new TextLoader.Range() { Min=0, Max=0} },
+                            Type = DataKind.Num
+                        },
+
+                        new TextLoader.Column()
+                        {
+                            Name = "Placeholder",
+                            Source = new [] { new TextLoader.Range() { Min=1, Max=784} },
+                            Type = DataKind.Num
+                        }
+                    }
+                }, new MultiFileSource(dataPath));
+
+                IDataView trans = TensorFlowTransform.Create(env, loader, model_location, "Softmax", "Placeholder");
+                trans = new ConcatTransform(env, trans, "reshape_input", "Placeholder");
+                trans = TensorFlowTransform.Create(env, trans, model_location, "dense/Relu", "reshape_input");
+                trans = new ConcatTransform(env, trans, "Features", "Softmax", "dense/Relu");
+
+                var trainer = new LightGbmMulticlassTrainer(env, new LightGbmArguments());
+
+                var cached = new CacheDataView(env, trans, prefetch: null);
+                var trainRoles = new RoleMappedData(cached, label: "Label", feature: "Features");
+                var pred = trainer.Train(trainRoles);
+
+                // Get scorer and evaluate the predictions from test data
+                IDataScorerTransform testDataScorer = GetScorer(env, trans, pred, testDataPath);
+                var metrics = Evaluate(env, testDataScorer);
+
+                Assert.Equal(0.99, metrics.AccuracyMicro, 2);
+                Assert.Equal(0.99, metrics.AccuracyMicro, 2);
```

Comment:
Is this an intentional duplicate line?

Reply:
Looks like a duplicate line. Addressed in follow up [PR](https://github.com/dotnet/machinelearning/pull/783)

---
In reply to: [214076676](https://github.com/dotnet/machinelearning/pull/704#discussion_r214076676) [](ancestors = 214076676)